### ISSUE TITLE
errors building latest ZFS against latest linux kernel; since check_disk_change seems to have disappeared and struct percpu_ref doesn't have a "count" member

### ISSUE THREAD

**Issue by superbaud** [openzfs/zfs#11074]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Arch linux
Distribution Version	| n/a rolling release
Linux Kernel	| v5.9-7810-g840e5bb326bb 
Architecture	| x86_64
ZFS Version	| zfs-2.0.0-rc1-139-g41e2b3de1
SPL Version	| zfs-2.0.0-rc1-139-g41e2b3de1

### Describe the problem you're observing
trying to build latest ZFS (on latest linux kernel) and getting multiple errors related to check_disk_change and struct percpu_ref.
### Describe how to reproduce the problem
checkout 41e2b3de13521230e3447e737638e73830ddfa3d from zfs repository (currently the latest commit) and try building it against a linux kernel with git hash 840e5bb326bbcb16ce82dd2416d2769de4839aea (currently the latest from the linux kernel repository), and there's multiple errors. i ran "make -k" to get all the errors to print out.

### Include any warning/errors/backtraces from the system logs

```
  CC [M]  /home/thz/zfs/module/zfs/../os/linux/zfs/vdev_disk.o
In file included from /home/thz/zfs/include/os/linux/spl/sys/uio.h:31,
                 from /home/thz/zfs/include/os/linux/spl/sys/sunddi.h:28,
                 from /home/thz/zfs/include/sys/zfs_context.h:69,
                 from /home/thz/zfs/module/zfs/../os/linux/zfs/vdev_disk.c:29:
/home/thz/zfs/module/zfs/../os/linux/zfs/vdev_disk.c: In function ‘vdev_disk_open’:
/home/thz/zfs/include/os/linux/kernel/linux/blkdev_compat.h:275:37: error: implicit declaration of function ‘check_disk_change’; did you mean ‘bdev_disk_changed’? [-Werror=implicit-function-declaration]
  275 | #define vdev_bdev_reread_part(bdev) check_disk_change(bdev)
      |                                     ^~~~~~~~~~~~~~~~~
/home/thz/zfs/module/zfs/../os/linux/zfs/vdev_disk.c:223:17: note: in expansion of macro ‘vdev_bdev_reread_part’
  223 |     int error = vdev_bdev_reread_part(bdev);
      |                 ^~~~~~~~~~~~~~~~~~~~~
/home/thz/zfs/module/zfs/../os/linux/zfs/vdev_disk.c: In function ‘vdev_blkg_tryget’:
/home/thz/zfs/module/zfs/../os/linux/zfs/vdev_disk.c:471:37: error: ‘struct percpu_ref’ has no member named ‘count’
  471 |   rc = atomic_long_inc_not_zero(&ref->count);
      |                                     ^~
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:283: /home/thz/zfs/module/zfs/../os/linux/zfs/vdev_disk.o] Error 1
  CC [M]  /home/thz/zfs/module/zfs/../os/linux/zfs/zvol_os.o
/home/thz/zfs/module/zfs/../os/linux/zfs/zvol_os.c: In function ‘zvol_open’:
/home/thz/zfs/module/zfs/../os/linux/zfs/zvol_os.c:504:2: error: implicit declaration of function ‘check_disk_change’; did you mean ‘bdev_disk_changed’? [-Werror=implicit-function-declaration]
  504 |  check_disk_change(bdev);
      |  ^~~~~~~~~~~~~~~~~
      |  bdev_disk_changed
/home/thz/zfs/module/zfs/../os/linux/zfs/zvol_os.c: In function ‘zvol_update_volsize’:
/home/thz/zfs/module/zfs/../os/linux/zfs/zvol_os.c:656:2: error: implicit declaration of function ‘revalidate_disk’; did you mean ‘zvol_revalidate_disk’? [-Werror=implicit-function-declaration]
  656 |  revalidate_disk(zv->zv_zso->zvo_disk);
      |  ^~~~~~~~~~~~~~~
      |  zvol_revalidate_disk
cc1: some warnings being treated as errors

```

**Comment by superbaud** [comment]
sorry for not including it in the above issue; but linux kernel commit bbf5c979011a099af5dc76498918ed7df445635b along with ZFS version: zfs-2.0.0-rc1_122_g61868bb14 zfs-kmod-2.0.0-rc1_122_g61868bb14 worked perfectly so the issue[s] were introduced (not sure if in ZFS or linux kernel) between those commits and those in the above issue.

**Comment by superbaud** [comment]
Fixed by #11085, thanks!
